index.bs

```diff
@@ -0,0 +1,550 @@
+
+This specification defines a [=policy-controlled feature=] identified by the
+string `"idle-detection"`. Its [=default allowlist=] is `["self"]`.
+
+<div class="note">
+The [=default allowlist=] of `["self"]` allows usage of this feature on
+same-origin nested frames by default but prevents access by third-party content.
+
+Third-party usage can be selectively enabled by adding the
+`allow="idle-detection"` attribute to an {{iframe}} element:
+
+<div class="example">
+```html
+<iframe src="https://example.com" allow="idle-detection"></iframe>
+```
+</div>
+
+Alternatively, this feature can be disabled completely in first-party contexts
+by specifying the permissions policy in an HTTP response header:
+
+<div class="example">
+```http
+Permissions-Policy: idle-detection 'none'
+```
+</div>
+
+See [[PERMISSIONS-POLICY]] for more details.
+</div>
+
+## The {{IdleDetector}} interface ## {#api-idledetector}
+
+<xmp class="idl">
+dictionary IdleOptions {
+  [EnforceRange] unsigned long threshold;
+  AbortSignal signal;
+};
+
+[
+  SecureContext,
+  Exposed=(Window,DedicatedWorker)
+] interface IdleDetector : EventTarget {
+  constructor();
+  readonly attribute UserIdleState? userState;
+  readonly attribute ScreenIdleState? screenState;
+  attribute EventHandler onchange;
+  [Exposed=Window] static Promise<PermissionState> requestPermission();
+  Promise<void> start(optional IdleOptions options = {});
+};
+</xmp>
+
+Instances of {{IdleDetector}} are created with the [=internal slots=] described
+in the following table:
+
+<table dfn-for=IdleDetector dfn-type=attribute>
+  <tr>
+    <th>[=Internal slot=]</th>
+    <th>Initial value</th>
+    <th>Description (non-normative)</th>
+  </tr>
+  <tr>
+    <td><dfn>\[[state]]</dfn></th>
+    <td>`"stopped"`</td>
+    <td>Tracks the active state of the {{IdleDetector}}</td>
+  </tr>
+  <tr>
+    <td><dfn>\[[threshold]]</dfn></th>
+    <td>`undefined`</td>
+    <td>The configured idle detection threshold</td>
+  </tr>
+</table>
+
+<wpt>
+  idlharness-worker.https.window.js
+  idlharness.https.window.js
+</wpt>
+
+### {{IdleDetector/userState}} attribute ### {#api-idledetector-userstate}
+
+The <dfn attribute for=IdleDetector>userState</dfn> attribute MUST be
+initialized to `null`.
+
+<div algorithm="user becomes idle">
+
+If |this|.{{[[state]]}} is `"started"`, |this|.{{userState}} is {{"active"}} and
+the user has not interacted with the device in at least |this|.{{[[threshold]]}}
+milliseconds, the [=user agent=] MUST [=queue a task=] to run the following
+steps:
+
+1.  Set |this|.{{userState}} to {{"idle"}}.
+1.  [=Fire an event=] named {{"change"}} using the {{Event}} interface at
+    |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+<div algorithm="user becomes active">
+
+If |this|.{{[[state]]}} is `"started"`, |this|.{{userState}} is {{"idle"}} and
+the user interacts with the device in, the [=user agent=] MUST [=queue a task=]
+to run the following steps:
```

Comment:
On what task source? Also it'd be ideal to specify an event loop and document, where applicable. We've recently made this a bit easier by introducing [queue a global task](https://html.spec.whatwg.org/#queue-a-global-task), so here you'd do

"Queue a task on the XXX task source with _idleDetector_'s relevant global object to perform these steps:"

Reply:
Which task source should I use? To resolve this and your previous comment about "this" I am thinking about something like,

For each instance of {{IdleDetector}} let it be |detector|, [=queue a task=] in its the [=relevant global object=]'s default task source to run these steps: